pages/advanced-algorithms/available-algorithms/algo.mdx

```diff
@@ -0,0 +1,442 @@
+
+
+{<h5> Usage with custom heuristic: </h5>}
+
+In this case, the default heuristic isn't used. Instead, each node already has manually calculated haversine distance,
+saved under `heur` property. Now, in the config map, a `heuristic_name` property must be included.
+
+```cypher
+CREATE (c1:City1 {heur: 157.2}), (c2:City2 {heur: 157.2}), (c3:City3 {heur: 157.2}),
+(c4:City4 {heur: 157.2}),
+(c5:City5 {heur: 0}),
+(m1:MO1 {heur: 78.62}), (m2:MO2 {heur: 78.62}),
+(m3:MO3 {heur: 78.62}) ,(m4:MO4 {heur: 78.62}),
+(c1)-[r1:ROAD {distance: 300}]->(c2),
+(c2)-[r2:ROAD {distance: 300}]->(c3),
+(c4)-[r3:ROAD {distance: 280}]->(c3),
+(c4)-[r4:ROAD {distance: 310}]->(c1),
+(c1)-[r5:RIVER {distance: 90}]->(m1),
+(c2)-[r6:ROAD {distance: 90}]->(m2),
+(c3)-[r7:ROAD {distance: 85}]->(m3),
+(c4)-[r8:ROAD {distance: 100}]->(m4),
+(m1)-[r9:RIVER {distance: 140}]->(m4),
+(m4)-[r10:RIVER {distance: 145}]->(m3),
+(m3)-[r11:RIVER {distance: 130}]->(m2),
+(m2)-[r12:RIVER {distance: 125}]->(m1),
+(m2)-[r13:ROAD {distance: 90}]->(c5),
+(m3)-[r14:RIVER {distance: 100}]->(c5)
+```
+
+```cypher
+MATCH (c1:City1 ), (c5:City5 )
+CALL algo.astar(c1,c5, {heuristic_name: "heur"}) YIELD path, weight RETURN path, weight;
+```
+
+Result:
+
+![](/pages/advanced-algorithms/available-algorithms/algo/astar_result6.png)
+
+```plaintext
++----------------------------+
+| weight                     |
++----------------------------+
+| 305                        |
++----------------------------+
+```
+
+{<h5> Usage with duration: </h5>}
+
+In this case, the heuristic value and distance value aren't `Numeric` data types, but rather `Duration` data types.
+Also, only `RIVER` relationships are allowed, both incoming and outgoing. Since a custom heuristic is used, it needs to be specified
+alongside other properties.
+
+```cypher
+CREATE (c1:City1 {heur: duration("PT1H30M")}), (c2:City2 {heur: duration("PT1H30M")}), (c3:City3 {heur: duration("PT1H30M")}),
+(c4:City4 {heur: duration("PT1H30M")}),
+(c5:City5 {heur: duration("PT0H")}),
+(m1:MO1 {heur: duration("PT47M")}), (m2:MO2 {heur: duration("PT47M")}),
+(m3:MO3 {heur: duration("PT47M")}) ,(m4:MO4 {heur: duration("PT47M")}),
+(c1)-[r1:ROAD {distance: duration("PT3H")}]->(c2),
+(c2)-[r2:ROAD {distance: duration("PT3H")}]->(c3),
+(c4)-[r3:ROAD {distance: duration("PT2H48M")}]->(c3),
+(c4)-[r4:ROAD {distance: duration("PT3H6M")}]->(c1),
+(c1)-[r5:RIVER {distance: duration("PT54M")}]->(m1),
+(c2)-[r6:ROAD {distance: duration("PT54M")}]->(m2),
+(c3)-[r7:ROAD {distance: duration("PT51M")}]->(m3),
+(c4)-[r8:ROAD {distance: duration("PT1H")}]->(m4),
+(m1)-[r9:RIVER {distance: duration("PT1H24M")}]->(m4),
+(m4)-[r10:RIVER {distance: duration("PT1H27M")}]->(m3),
+(m3)-[r11:RIVER {distance: duration("PT1H18M")}]->(m2),
+(m2)-[r12:RIVER {distance: duration("PT1H15M")}]->(m1),
+(m2)-[r13:ROAD {distance: duration("PT54M")}]->(c5),
+(m3)-[r14:RIVER {distance: duration("PT1H")}]->(c5)
+
+```
+
```

Comment:
```suggestion
As the node is not using the default names for the heuristic properties, and they, as well as distances, are of type `Duration` rather than the default `Number` data type, they need to be specified in the configuration map. 
Also, the query searches for a path using only `RIVER` relationships, both incoming and outgoing.
alongside other properties.

```